### ISSUE TITLE
Add support for fully randomized test runs locally

### ISSUE THREAD

**Issue by bradfrizzell** [ampproject/amphtml#10448]
Our testing infrastructure, when run locally, should run all tests in random order to facilitate detection of flaky tests / tests that are leaking state. 

I've already added support for running tests in random order by file (i.e. the tests within each file still run from top to bottom). To do this, run `gulp test --randomize`

However, if we can modify the test runner to fully randomize, that would be a further benefit. 

The PR that added random by file is https://github.com/ampproject/amphtml/pull/8779

**Comment by rsimha-amp** [comment]
Per-file randomization of test execution isn't currently supported by mocha, but there is (was?) an ongoing effort to make it work. See https://github.com/mochajs/mocha/issues/902

Until then, I don't think there's much we can do here.

**Comment by jridgewell** [comment]
We can override `describe`, `beforeEach`, `afterEach`, and `it` blocks, create the tree structure, randomize it, and feed that to Mocha.

**Comment by rsimha-amp** [comment]
There might be an easier way: https://www.npmjs.com/package/rocha

**Comment by rsimha-amp** [comment]
@erwinmombay, I'm going to take a shot at adding a way to use rocha with karma. Not sure if it's possible or not.

**Comment by rsimha-amp** [comment]
Unfortunately, there's no `karma-rocha` bridge similar to https://github.com/karma-runner/karma-mocha, so rocha looks like a non-starter for our purposes.

**Comment by rsimha-amp** [comment]
Since we primarily want randomized testing for local development, it should be possible to achieve it by running mocha / rocha from the command line. Currently, it appears that this is blocked by #9695, after which we should have babel 6 support. Right now, running mocha / rocha from the command line returns `SyntaxError: Unexpected token import`.

I'll revisit this issue once we have babel 6 support.

**Comment by bradfrizzell** [comment]
Now that we have babel 6, will Rocha work?

**Comment by rsimha-amp** [comment]
Unfortunately, after some testing post-babel-6, `rocha` is looking like a dead end too, given how different its API is from mocha, and due to the fact that we use `karma-mocha-runner` to run our tests.

Until https://github.com/mochajs/mocha/issues/902 is fixed, I'm afraid this issue is a won't fix.
